index.js

```diff
@@ -190,10 +190,12 @@ module.exports = {
     let addonProvidedConfig = this._getAddonProvidedConfig(config);
     let shouldCompileModules = this._shouldCompileModules(config);
 
-    let providedAnnotation = config['ember-cli-babel'] && config['ember-cli-babel'].annotation;
+    let addonOptions = config['ember-cli-babel'] || {};
+    let babelOptions = config.babel || {};
 
     let options = {
-      annotation: providedAnnotation || `Babel: ${this._parentName()}`
+      annotation: addonOptions.annotation || `Babel: ${this._parentName()}`,
+      sourceMaps: addonOptions.sourceMaps || babelOptions.sourceMaps
```

Comment:
Both are still valid, but each of them for different types of options.

At this point, I would say that `sourceMaps` should go in `babel` config (since we do pass it through directly to babel itself), and that we shouldn't allow it to be specified in `options['ember-cli-babel']`.